server/reload_test.go

```diff
@@ -4155,3 +4155,156 @@ func TestReloadValidate(t *testing.T) {
 	}
 	srv.Shutdown()
 }
+
+func TestConfigReloadAccounts(t *testing.T) {
+	conf := createConfFile(t, []byte(`
+	listen: "127.0.0.1:-1"
+	system_account: SYS
+	accounts {
+		SYS {
+			users = [
+				{user: sys, password: pwd}
+			]
+		}
+		ACC {
+			users = [
+				{user: usr, password: pwd}
+			]
+		}
+		acc_deleted_after_reload_will_trigger_reload_of_all_accounts {
+			users = [
+				{user: notused, password: soon}
+			]
+		}
+	}
+	`))
+	defer os.Remove(conf)
+	s, o := RunServerWithConfig(conf)
+	defer s.Shutdown()
+
+	urlSys := fmt.Sprintf("nats://sys:pwd@%s:%d", o.Host, o.Port)
+	urlUsr := fmt.Sprintf("nats://usr:pwd@%s:%d", o.Host, o.Port)
+	oldAcc, ok := s.accounts.Load("SYS")
+	if !ok {
+		t.Fatal("No SYS account")
+	}
+
+	testSrvState := func(oldAcc interface{}) {
+		sysAcc := s.SystemAccount()
+		s.mu.Lock()
+		fatal := func() string {
```

Comment:
You should have the defer s.mu.Unlock() for this test function. If there is an error, you would return without the unlock and the test would probably "stall" due to a deadlock (not reporting the actual error)

Reply:
I used a closure so this wouldn't happen, but you are right, defer is much better.